### ISSUE TITLE
AlphaPose/detector/nms/src/nms_kernel.cu:5:10: fatal error: THC/THC.h: No such file or directory     5 | #include <THC/THC.h>

### ISSUE THREAD

**Issue by jinfagang** [MVIG-SJTU/AlphaPose#980]
For decent pytorch version 1.11 (maybe 1.10), this file removed:

```
AlphaPose/detector/nms/src/nms_kernel.cu:5:10: fatal error: THC/THC.h: No such file or directory
    5 | #include <THC/THC.h>

```

![image](https://user-images.githubusercontent.com/21303438/151491285-f26ec912-964c-426b-8907-5d46991c71cd.png)

**Comment by Fang-Haoshu** [comment]
Oh... Did you manage to fix it?

**Comment by jinfagang** [comment]
No, I suggest using torchvision.ops.boxes.nms to do nms operation, however, which needs modification on detector. But since Alphapose already included that source code, it also worthy to modificate to standard way

**Comment by jinfagang** [comment]
Please using my fork version: it works on torch 1.11

https://github.com/jinfagang/AlphaPose

**Comment by peteruhrig** [comment]
And it turns out that this is a real show-stopper for A40 and A100 cards. They require CUDA 11.1 or higher, thus PyTorch 1.11 or higher, which does not contain the THC component any more. Thus, as of now, I don't see how to run AlphaPose on recent cards.

**Comment by cugcuiyc** [comment]
> 

@jinfagang  Still the same error. You didn't modify the code in `detector/nms/src/nms_kernel.cu` at branch master.

**Comment by cugcuiyc** [comment]
> And it turns out that this is a real show-stopper for A40 and A100 cards. They require CUDA 11.1 or higher, thus PyTorch 1.11 or higher, which does not contain the THC component any more. Thus, as of now, I don't see how to run AlphaPose on recent cards.

@peteruhrig Any good suggestions?

**Comment by peteruhrig** [comment]
I don't speak C/C++ well enough, but my understanding is that it is possible to replace the functions used with ones that are still supported, as discussed here:
https://www.exxactcorp.com/blog/Deep-Learning/pytorch-1-11-0-now-available
(scroll down to the CUDA section)
My guess is that this is not very difficult for someone who knows what they are doing, and maybe that is what @jinfagang did but forgot to push back to his repo, but I don't understand it enough to be able to do this myself.
